includes/options/class-amp-options-rest-controller.php

```diff
@@ -0,0 +1,150 @@
+				],
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'update_items' ],
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+					'permission_callback' => [ $this, 'get_items_permissions_check' ],
+				],
+				'schema' => $this->get_public_item_schema(),
+			]
+		);
+	}
+
+	/**
+	 * Checks whether the current user has permission to manage options.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has permission; WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return new WP_Error(
+				'amp_rest_cannot_manage_options',
+				__( 'Sorry, you are not allowed to manage options for the AMP plugin for WordPress.', 'amp' ),
+				[ 'status' => rest_authorization_required_code() ]
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves all AMP plugin options.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$options    = AMP_Options_Manager::get_options();
+		$properties = $this->get_item_schema()['properties'];
+
+		$options = wp_array_slice_assoc( $options, array_keys( $properties ) );
+		return rest_ensure_response( $options );
+	}
+
+	/**
+	 * Updates AMP plugin options.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return array|WP_Error Array on success, or error object on failure.
+	 */
+	public function update_items( $request ) {
+		$params = $request->get_params();
+
+		AMP_Options_Manager::update_options( wp_array_slice_assoc( $params, array_keys( $this->get_item_schema()['properties'] ) ) );
+
+		return rest_ensure_response( $this->get_items( $request ) );
+	}
+
+	/**
+	 * Retrieves the schema for plugin options provided by the endpoint.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( ! $this->schema ) {
+			$this->schema = [
+				'$schema'    => 'http://json-schema.org/draft-04/schema#',
+				'title'      => 'amp-wp-options',
+				'type'       => 'object',
+				'properties' => [
+					// Note: The sanitize_callback from AMP_Options_Manager::register_settings() is applying to this option.
+					Option::THEME_SUPPORT => [
+						'type' => 'string',
+						'enum' => [
+							AMP_Theme_Support::READER_MODE_SLUG,
+							AMP_Theme_Support::STANDARD_MODE_SLUG,
+							AMP_Theme_Support::TRANSITIONAL_MODE_SLUG,
+						],
+					],
+					Option::READER_THEME  => [
+						'type'        => 'string',
+						'arg_options' => [
+							'validate_callback' => function ( $value ) {
```

Comment:
```suggestion
							'validate_callback' => static function ( $value ) {
```

Reply:
Same as above, I think we need the `$this` in the callback.